File: BE/src/main/java/com/example/beprojectweb/entity/Order.java
package com.example.beprojectweb.entity;

import com.example.beprojectweb.enums.OrderStatus;
import com.example.beprojectweb.enums.PaymentMethod;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
@Table(name = "orders")
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    User user;

    BigDecimal totalPrice;            // Tổng tiền đơn hàng

    @Enumerated(EnumType.STRING)
    OrderStatus status;               // Trạng thái (PENDING, PAID, COMPLETED, CANCELLED)

    @Enumerated(EnumType.STRING)
    @Column(name = "payment_method")
    PaymentMethod paymentMethod;             // Hình thức thanh toán (COD, MOMO, ZaloPay...)

    LocalDateTime createdAt;          // Thời gian đặt
    LocalDateTime paymentAt;          // Thời gian thanh toán (nếu có)

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
    Set<OrderItem> orderItems = new HashSet<>(); // Danh sách sản phẩm
}

File: FE/src/model/Order.ts
// Model cho sản phẩm trong đơn hàng
export interface OrderItem {
    id: number;           // ID của OrderItem
    productId: number;    // ID sản phẩm
    productName: string;  // Tên sản phẩm
    totalPrice: number;   // Giá tại thời điểm mua
}

// Model chính cho đơn hàng
export interface Order {
    id: number;                 // ID đơn hàng
    userId: number;             // ID người dùng đặt hàng
    totalPrice: number;         // Tổng tiền đơn hàng
    status: string;             // Trạng thái đơn hàng (ví dụ: "PENDING", "PAID", ...)
    paymentMethod?: string;     // Hình thức thanh toán (ví dụ: "COD", "MOMO")
    createdAt: string;          // Thời gian tạo đơn (ISO String)
    paymentAt?: string;         // Thời gian thanh toán (ISO String), có thể undefined nếu chưa thanh toán
    orderItems: OrderItem[];    // Danh sách sản phẩm trong đơn hàng
}

// Dữ liệu gửi lên backend khi tạo hoặc cập nhật đơn hàng
export interface OrderRequest {
    paymentMethod?: string;  // Hình thức thanh toán, nếu có
}

// Dữ liệu backend trả về khi lấy chi tiết đơn hàng
export interface OrderResponse {
    id: number;
    userId: number;
    totalPrice: number;
    status: string;
    paymentMethod?: string;
    createdAt: string;
    paymentAt?: string;
    orderItems: OrderItem[];
}

export class PaymentMethodResponse {
}